clang/test/ParserHLSL/hlsl_resource_class_attr_error.hlsl

```diff
@@ -7,7 +7,7 @@ struct [[hlsl::resource_class()]] Eg1 {
 
 Eg1 e1;
 
-// expected-error@+1{{invalid resource class 'gibberish' used; expected 'SRV', 'UAV', 'CBuffer', or 'Sampler'}}
+// expected-warning@+1{{ResourceClass attribute argument not supported: gibberish}}

```

Comment:
So as a warning....how is this interpreted?  Is the entire attribute ignored?  So at some point when something requires that attribute to be present we'll get an error?

Reply:
The attribute never gets added, so yes, it is accurate to view this warning as a declaration that the attribute is ignored. This is the exact same behavior as attaching an HLSL Shader attribute with an invalid argument.
If something requires the attribute to be present, an error would be emitted, but if there is no such requirement, I think compilation could succeed.